### DISCUSSION TITLE
How to display the cell for PolyData?

### DISCUSSION THREAD

**Discussion by zhang-qiang-github** [pyvista/pyvista#3819] [Q&A]
Please have a look for following code:

```
import pyvista as pv

sphere = pv.Sphere()
print(sphere)
```
And the printed information is:

```
PolyData (0x262e5cb6d60)
  N Cells:	1680
  N Points:	842
  X Bounds:	-4.993e-01, 4.993e-01
  Y Bounds:	-4.965e-01, 4.965e-01
  Z Bounds:	-5.000e-01, 5.000e-01
  N Arrays:	1
```

I want to know: how to display the exact value for 842 points and 1680 cells?

Actually, I have looked at the source code, the `pv.Sphere()` firstly generate the sphere using `vtkSphereSource`, and wrap it to `PolyData` by `pyvista.wrap(sphere.GetOutput())`. The call stack is:

```
PolyData.__init__
_PointSet.shallow_copy
DataObject.shallow_copy
```
In the `DataObject.shallow_copy`, it call `self.ShallowCopy(to_copy)`. Then, I cannot step into `ShallowCopy`, and I don't know how it deal with `vtkPolyData`.

How can I print the exact value for `PolyData` (`points&cells`)? How `pyvista` deal with `vtkPolyData` in `ShallowCopy`? Any suggestion is appreciated~~~

**Comment by MatthewFlamm** [comment]
What do you mean by

> How can I print the exact value for PolyData (points&cells)? 

Do you mean the location in (x, y, z) of the points?  This would be `sphere.points`.

**Reply by zhang-qiang-github** [reply]
Thanks for kindly reply. How about `cells`? How to print the value of `cells`?

**Reply by MatthewFlamm** [reply]
The cells are more complex.  They are defined usually by a set of points and a shape.  You can get the cell centers with `sphere.cell_centers().points`, but this is not really THE cell.

**Reply by MatthewFlamm** [reply]
See https://docs.pyvista.org/user-guide/what-is-a-mesh.html#what-is-a-mesh

**Reply by zhang-qiang-github** [reply]
I can only obtain the cell by vtk code:

```
N = polydata.GetNumberOfCells()
for j in range(N):
    cell = polydata.GetCell(j)
    id0 = cell.GetPointIds().GetId(0)
    id1 = cell.GetPointIds().GetId(1)
    id2 = cell.GetPointIds().GetId(2)
    conn.append([id0, id1, id2])
```
But, it is inconvenient. Hopes `pyvista` could provide some convenient method.

**Reply by MatthewFlamm** [reply]
If you want the points of a cell use `sphere.cell_points` or `sphere.cell_point_ids` in currently released versions.  In the next minor release, you would use `sphere.get_cell(0).points` and `sphere.get_cell(0).point_ids`.

**Reply by zhang-qiang-github** [reply]
> If you want the points of a cell use `sphere.cell_points` or `sphere.cell_point_ids` in currently released versions. In the next minor release, you would use `sphere.get_cell(0).points` and `sphere.get_cell(0).point_ids`.

For `sphere.cell_point_ids`, it reported `PolyData` object has no attribute `cell_point_ids`. My `pyvista` version is `0.34.1`.

**Reply by MatthewFlamm** [reply]
This was added in 0.35.1 https://github.com/pyvista/pyvista/releases/tag/v0.35.1
